src/services/sablier.js

```diff
@@ -283,21 +223,8 @@ export const getCreateStreamFeeAndTransaction = async (
     symbol: ETH,
   };
 
-  if (!approvedAmountBN || amount.gt(approvedAmountBN)) {
-    const rawValue = 1000000000;
-    const valueToApprove = utils.parseUnits(rawValue.toString(), asset.decimals);
-    const approveTransactionData = buildERC20ApproveTransactionData(
-      getEnv().SABLIER_CONTRACT_ADDRESS, valueToApprove, asset.decimals);
-
-    sablierCreateStreamTransaction = {
-      from: sender,
-      to: asset.address,
-      data: approveTransactionData,
-      amount: 0,
-      symbol: ETH,
-      sequentialSmartWalletTransactions: [sablierCreateStreamTransaction],
-    };
-  }
+  sablierCreateStreamTransaction = await addAllowanceTransaction(
+    sablierCreateStreamTransaction, sender, getEnv().SABLIER_CONTRACT_ADDRESS, asset, amount);
```

Comment:
The transaction will fail if the required amount is already allowed.

Reply:
The function `addAllowanceTransaction` checks for allowance and adds allowance transaction only when it's needed
And btw why would it fail? Afaik you can ask for allowance multiple times for whatever amount you want.

Reply:
The estimation will fail and then transaction.